test/source-updater.test.js

```diff
@@ -1,232 +1,769 @@
-  let sourceBuffer;
+  const done = assert.async();
+
+  // Source buffer must exist for the callback to run. This case isn't tested, as it isn't
+  // required behavior (at the moment), but is necessary to know for this test.
+  this.sourceUpdater.createSourceBuffers({
+    video: 'avc1.4D001E'
+  });
 
-  updater.appendBuffer(new Uint8Array([0, 1, 2]));
+  let executedCallback = false;
+  let appendedVideo = false;
 
-  this.mediaSource.trigger('sourceopen');
-  sourceBuffer = this.mediaSource.sourceBuffers[0];
-  assert.equal(sourceBuffer.updates_.length, 0, 'did not call the source buffer');
+  this.sourceUpdater.appendBuffer('video', mp4Video(), () => {
+    appendedVideo = true;
+    assert.notOk(executedCallback, 'haven\'t executed callback');
+    setTimeout(() => {
+      assert.ok(executedCallback, 'executed callback');
+      done();
+    }, 0);
+  });
 
-  // create other media source
-  this.mediaSource.addSourceBuffer('audio/mp2t');
-  sourceBufferEmitter.trigger('sourcebufferadded');
+  assert.notOk(appendedVideo, 'haven\'t appended video before callback is queued');
 
-  assert.equal(sourceBuffer.updates_.length, 1, 'called the source buffer once');
-  assert.deepEqual(sourceBuffer.updates_[0].append, new Uint8Array([0, 1, 2]),
-                  'appended the bytes');
+  this.sourceUpdater.videoQueueCallback(() => {
+    executedCallback = true;
+  });
 });
 
-QUnit.test('runs the completion callback when updateend fires', function(assert) {
-  let updater = new SourceUpdater(this.mediaSource, 'video/mp2t');
-  let updateends = 0;
-  let sourceBuffer;
+QUnit.test('audioQueueCallback does not call video queue callback after queue empties',
+function(assert) {
+  const done = assert.async();
 
-  this.mediaSource.trigger('sourceopen');
-  sourceBuffer = this.mediaSource.sourceBuffers[0];
-  updater.appendBuffer(new Uint8Array([0, 1, 2]), function() {
-    updateends++;
+  // Source buffer must exist for the callback to run. This case isn't tested, as it isn't
+  // required behavior (at the moment), but is necessary to know for this test.
+  this.sourceUpdater.createSourceBuffers({
+    audio: 'mp4a.40.2'
   });
-  updater.appendBuffer(new Uint8Array([2, 3, 4]), function() {
-    throw new Error('Wrong completion callback invoked!');
+
+  let executedCallback = false;
+  let appendedAudio = false;
+
+  this.sourceUpdater.appendBuffer('audio', mp4Audio(), () => {
+    appendedAudio = true;
+    assert.notOk(executedCallback, 'haven\'t executed callback');
+    setTimeout(() => {
+      assert.notOk(executedCallback, 'haven\'t executed callback');
+      done();
+    }, 0);
   });
 
-  assert.equal(updateends, 0, 'no completions yet');
-  sourceBuffer.trigger('updateend');
-  assert.equal(updateends, 1, 'ran the completion callback');
+  assert.notOk(appendedAudio, 'haven\'t appended audio before callback is queued');
+
+  this.sourceUpdater.videoQueueCallback(() => {
+    executedCallback = true;
```

Comment:
this is what we _don't_ want to happen, right? Is the test name incorrect?

Reply:
Yup, test name should be correct, though I can change the variable name to `executedVideoCallback` to be more precise.